test/jasmine/tests/axes_test.js

```diff
@@ -8217,3 +8216,65 @@ describe('shift tests', function() {
         });
     });
 });
+describe('test tickmode calculator', function() {
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    function generateTickConfig(){
+      standardConfig = {tickmode: 'array', ticks: 'inside', ticklen: 1, showticklabels: false};
+
+      // Number of ticks will be random
+      var n = Math.floor(Math.random() * 99) + 1;
+      tickVals = [];
+      for(let i = 0; i <= n; i++) tickVals.push(i);
+      standardConfig['tickvals'] = tickVals;
+      standardConfig['ticktext'] = tickVals;
+      return standardConfig;
+    }
+    var ticksOff = {tickmode:"array", ticks: '', tickvals:[], ticktext:[], ticklen: 0, showticklabels: false};
+    // the goal is target the arrayTicks() function, the subject of PR: https://github.com/plotly/plotly.js/pull/6829 
+    // we test xMajor and xMinor in on/on on/off off/on and off/off
+    // since we can't unit test functions that are not exported, we shim functions we don't care about instead and
+    // test the nearest exported functions (calcTicks)
+    describe('arrayTicks', function() {
+        for(let i = 0; i < 4; i++) {
+            (function(i) {
```

Comment:
Please declare this function outside the loop and use ES5 to pass the syntax test.